pmd-core/src/main/java/net/sourceforge/pmd/renderers/internal/sarif/SarifLog.java

```diff
@@ -664,6 +699,11 @@ public boolean equals(final java.lang.Object o) {
             if (this$priority == null ? other$priority != null : !this$priority.equals(other$priority)) {
                 return false;
             }
+            final java.lang.Object this$tags = this.getTags();
```

Comment:
This whole green block could be `return Objects.equals(this.tags, other.tags);`. The rest of the equals could be refactored to use that too, it's hard to read and uses unconventional names.

Reply:
It looked like this was an auto-generated class so I was just keeping it consistent. It's the same throughout all the equal methods in this class and not sure they're even being used. What would you recommend in this instance?

Reply:
Actually I think you're right. This could be cleaned up but it doesn't need to be in this pr.